src/index.js

```diff
@@ -609,7 +613,11 @@ function createConfig(options, entry, format, writeMeta) {
 								if (writeMeta && nameCache) {
 									fs.writeFile(
 										getNameCachePath(),
-										JSON.stringify(nameCache, null, 2),
+										JSON.stringify(
+											endsWithNewLine ? `${nameCache}\r\n` : nameCache,
```

Comment:
Might be worth grabbing the trailing `\r?\n` from the original file so that the written result uses the same line endings.

Reply:
Good suggestion, I'll update it asap.